lua/hlargs/config.lua

```diff
@@ -32,7 +32,11 @@ local defaults = {
 
 function M.setup(opts)
   M.opts = vim.tbl_deep_extend("force", {}, defaults, opts or {})
-  vim.cmd("highlight def Hlargs guifg=" .. M.opts.color)
+  if opts and opts.highlight then
+    M.opts.highlight = opts.highlight
+  end
+  M.opts.highlight.default = true
```

Comment:
btw I don't think this is necessary, `tbl_deep_extend` should do that for you

Reply:
Nope. This is necessary because `opts.highlight` will be `{ fg = '#000000', link = 'TSParameter' }` if my config is `{ fg = '#000000' }`, which doesn't do the right thing.